src/core/annotation.js

```diff
@@ -1371,26 +1381,52 @@ class WidgetAnnotation extends Annotation {
     return [initialState.font, initialState.fontName, initialState.fontSize];
   }
 
-  _computeFontSize(font, fontName, fontSize, height) {
+  _computeFontSize(font, fontName, fontSize, height, lineCount) {
     if (fontSize === null || fontSize === 0) {
-      const em = font.charsToGlyphs("M")[0].width / 1000;
-      // According to https://en.wikipedia.org/wiki/Em_(typography)
-      // an average cap height should be 70% of 1em
-      const capHeight = 0.7 * em;
-      // 1.5 * capHeight * fontSize seems to be a good value for lineHeight
-      fontSize = Math.max(1, Math.floor(height / (1.5 * capHeight)));
+      // A zero value for size means that the font shall be auto-sized:
+      // its size shall be computed as a function of the height of the
+      // annotation rectangle (see 12.7.3.3).
```

Comment:
Shouldn't auto-sized also be calculated to fit all text inside the box? So if the text would be too long to fit inside the box, the font-size should be calculated to fit.

Reply:
AFAIK there is nothing about that in the pdf spec but pdfium is doing that so maybe we should too, but in an other patch.